### ISSUE TITLE
carta_backend built with ASAN flags crashes when opening closing a file during animation playback

### ISSUE THREAD

**Issue by ajm-ska** [CARTAvis/carta-backend#1052]
**Describe the bug**
If built with the ASAN flags, the carta_backend will occasionally crash if closing a file during animation playback. 
It was noticed to sometimes occur during the `CLOSE_FILE_ANIMATION.test.ts` ICD test on RHEL7 and Ubuntu 20.04. I have been to reproduce it manually on macOS12 while using the ASAN suppression file.

**To Reproduce**
1. Build the carta_backend with the following flags:
```
cmake .. -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_FLAGS='-O0 -g -fsanitize=address -fno-omit-frame-pointer' -DCMAKE_EXE_LINKER_FLAGS='-fsanitize=address' 
```
2. Start the carta_backend as you would normally do with the suppression file:
```
ASAN_OPTIONS=suppressions=/Users/ajm/carta-backend/debug/asan/myasan.supp ./carta_backend --frontend_folder=/Users/ajm/carta-frontend/build --verbosity=5
```
4. Open the `M17_SWex.fits` test image.
5. Start animation playback.
6. Close the image (Alt+w is handy).
7. Repeat steps 5 and 6 until the crash occurs.
(It is rare. In this case, I need to do it 10 times before I got the crash event).

**Expected behavior**
carta_backend should not crash.

**Screenshots or videos**
Here is the end of the crash log: 
[asan-animator-playback2.log](https://github.com/CARTAvis/carta-backend/files/8243277/asan-animator-playback2.log)

**Platform info (please complete the following information):**
 First found in ICD tests on Ubuntu 20.04 and RHEL7, but I can also trigger it manually on macOS12 (while using the ASAN suppression file), so it may be possible to occur on all platforms.
 - OS: macOS12
 - Browser Firefox
 - Version 98

**Additional context**
The crash only occurs when using the ASAN flags. Previously the `CLOSE_FILE_ANIMATION.test.ts` has been running without problems in CI for several months.

**Comment by jolopezl** [comment]
Likely related to [#1019](https://github.com/CARTAvis/carta-backend/issues/1019).

**Comment by ajm-ska** [comment]
This issue appears to be gone. It was most likely fixed in the 3.4.0+6.5.0+2022.4.12 carta-casacore update. I have not noticed it occurring in the ICD tests recently and I can not trigger it manually in macOS12. I also ran the `CLOSE_FILE_ANIMATION.test.ts` test in a RHEL7 Docker container 100 times and it passed every time. I feel we can close this issue now.
